go/vt/vttablet/faketmclient/fake_client.go

```diff
@@ -281,8 +281,8 @@ func (client *FakeTabletManagerClient) SlaveWasRestarted(ctx context.Context, ta
 }
 
 // StopReplicationAndGetStatus is part of the tmclient.TabletManagerClient interface.
-func (client *FakeTabletManagerClient) StopReplicationAndGetStatus(ctx context.Context, tablet *topodatapb.Tablet) (*replicationdatapb.Status, error) {
-	return &replicationdatapb.Status{}, nil
+func (client *FakeTabletManagerClient) StopReplicationAndGetStatus(ctx context.Context, tablet *topodatapb.Tablet, stopIOThreadOnly bool) (*replicationdatapb.Status, *replicationdatapb.StopSlaveStatus, error) {
+	return &replicationdatapb.Status{}, &replicationdatapb.StopSlaveStatus{}, nil
```

Comment:
Advice: I've been on that path; at some point in the future you may want other flags, too. Like: stop replication such that relay logs are fully consumed (therefore, stop IO thread, wait for SQL thread to catch up, stop SQL thread).
Merely a suggestion: instead of `stopIOThreadOnly` argument, provide a new `StopReplicationType` which can be something like `BothThreads, IOThreadOnly, MatchPositions` etc.

Reply:
I think that's a good point, although I think `BothThreads` should be the default. If you supply nil for `StopReplicationOptions` (proposed name) then that means `BothThreads` because that's the natural result of calling `stop slave`. What do you mean by `MatchPositions`?

Reply:
Agreed, `BothThreads` is the simplest and should be default. `MatchPositions` is just a bad name for a scenario where you want to stop IO thread and then run SQL thread until is has consumed all the remaining relay logs.

Reply:
I like the idea to change `stopIOThreadOnly` to an enum.